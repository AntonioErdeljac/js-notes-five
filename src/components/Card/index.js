import PropTypes from 'prop-types';
import React, { memo } from 'react';

const Card = ({ children, onClick }) => {
  return (
    <div
      onClick={onClick}
      className="col-12 col-sm-6 col-md-4 col-xl-2 d-inline-flex align-items-center justify-content-center py-5"
    >
      {children}
    </div>
  );
};

Card.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
};

Card.defaultProps = {
  onClick: null,
};

export default memo(Card);
