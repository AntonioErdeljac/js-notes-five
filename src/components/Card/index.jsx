import PropTypes from 'prop-types';
import React, { memo } from 'react';

const Card = ({ children }) => {
  return (
    <div className="col-12 col-sm-6 col-md-4 col-xl-3 d-inline-flex align-items-center justify-content-center py-5">
      {children}
    </div>
  );
};

Card.propTypes = {
  children: PropTypes.node.isRequired,
};

export default memo(Card);
