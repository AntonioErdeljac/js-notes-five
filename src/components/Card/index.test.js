import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Card from './index';

const makeProps = ({ children } = {}) => ({
  children: children || 'Content',
});

const renderComponent = (props) => shallow(<Card {...props} />);

describe('Card Component', () => {
  it('renders correctly according to snapshot', () => {
    expect(toJson(renderComponent(makeProps()))).toMatchSnapshot();
  });
});
