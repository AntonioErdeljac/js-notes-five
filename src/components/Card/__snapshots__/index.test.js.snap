// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly according to snapshot 1`] = `
<div
  className="col-12 col-sm-6 col-md-4 col-xl-3 d-inline-flex align-items-center justify-content-center py-5"
>
  Content
</div>
`;
